cloud-sql-java-functions-benchmark/src/main/java/functions/Main.java

```diff
@@ -0,0 +1,185 @@
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.sql.DataSource;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+
+public class Main implements HttpFunction {
+
+  // TODO: replace these values before deploying
+  private static final String PROJECT_ID = "";
+  private static final String CONNECTION_NAME = "";
+  private static final String DB_NAME = "";
+  private static final String DB_USER = "";
+  private static final String DB_PASSWORD = "";
+
+  private static final Logger LOGGER = Logger.getLogger(Main.class.getName());
+  private static final Tracer TRACER = Tracing.getTracer();
+
+  static {
+    try {
+      // Set up StackDriver
+      StackdriverTraceExporter.createAndRegister(
+          StackdriverTraceConfiguration.builder()
+              .setProjectId(PROJECT_ID)
+              .build());
+
+    } catch (java.io.IOException e) {
+      LOGGER.log(Level.WARNING, "Warning: Could not set up Stackdriver Trace", e);
+    }
+
+  }
+
+  @Override
+  public void service(HttpRequest request, HttpResponse response)
+      throws IOException {
+
+    Integer duration = Integer.parseInt(request.getFirstQueryParameter("duration").orElse("5000"));
+    Integer interval = Integer.parseInt(request.getFirstQueryParameter("interval").orElse("500"));
+
+    // Trace every request
+    TraceConfig traceConfig = Tracing.getTraceConfig();
+    traceConfig.updateActiveTraceParams(
+        traceConfig.getActiveTraceParams().toBuilder().setSampler(Samplers.alwaysSample()).build());
+
+    // Set up URL parameters
+    String jdbcURL = String.format("jdbc:postgresql:///%s", DB_NAME);
+    Properties connProps = new Properties();
+    connProps.setProperty("user", DB_USER);
+    connProps.setProperty("password", DB_PASSWORD);
+    connProps.setProperty("socketFactory", "com.google.cloud.sql.postgres.SocketFactory");
+    connProps.setProperty("cloudSqlInstance", CONNECTION_NAME);
+
+    LOGGER.log(Level.INFO, "Started logging.");
+
+    LOGGER.log(Level.INFO, "Started logging.");
+    // Create thread pool
+    ScheduledExecutorService execService = Executors.newScheduledThreadPool(2);
+
+    // Start tests for regular connections
+    execService.scheduleAtFixedRate(() -> {
+      connectRegular(jdbcURL, connProps);
+    }, 0, interval, TimeUnit.MILLISECONDS);
+
+    // Start tests for pooled connections
+    DataSource pool = createPool(jdbcURL, connProps);
+    execService.scheduleAtFixedRate(() -> {
+      connectWithPool(pool);
+    }, 0, interval, TimeUnit.MILLISECONDS);
+
+    try {
+      TimeUnit.MILLISECONDS.sleep(duration);
+    } catch (InterruptedException e) {
+      response.setStatusCode(HttpURLConnection.HTTP_INTERNAL_ERROR);
+      BufferedWriter writer = response.getWriter();
+      writer.write(ExceptionUtils.getStackTrace(e));
+      return;
+    }
+
+    execService.shutdown();
+
+    BufferedWriter writer = response.getWriter();
+    writer.write(
```

Comment:
nit: Might be nice to return a JSON response with some averages or something. Not sure if it's possible to get the info from TRACE before it exports.

Reply:
This might be something to look into later. I think since we added the UUID to the spans, we can get all the spans for the specific request, but I'm also not sure if all the info will be exported to Stackdriver before the request ends.